scripts/gen.py

```diff
@@ -819,6 +834,9 @@ def generate_aten_remap(ctx, fname, sig, params, fnopts):
     assert fnopts.outfn_name
     fcall = create_call(fnopts.outfn_name, param_vars)
 
+  if fnopts.shape_check:
```

Comment:
No, `__or__` isn't supposed to do a check, it's supposed to expand to a compatible shape. What is your suggestion here?

Reply:
How is ```s__th_or``` any different?
Delving into the maze of TH macros, it seems the same.
I'd be inclined in removing the shape_check from this code generation.

Reply:
No idea what the `s__` stands for, but it's supposed to enforce identical shapes for the two operators, whereas the non-`s__` versions expand to the same size. The check is part of the behavior, it's not redundant.

Reply:
Oh, I see. The magic APPLY3 macros does sizes enforcing.
Then we should have shape_check be a nested tuple with all the ordinals of the shapes to be checked, instead of blindly assuming 0 and 1.
Like:
```
shape_check=((0, 1),)
```

Reply:
Thought about it, but it's just a handful of quirky binary operators. In practice that information would always be `(0, 1)`.

Reply:
I already like this very little, but having a shape_check with bolted in 0,1 semantics adds to the worse.

Reply:
Not sure why we'd like better additional flexibility for something which we shouldn't be there in the first place and we don't control, but fine, I'll do it.

Reply:
If it should not be there we should remove it.
But since it seems those APIs carry a size-check semantics, there is no reason to add one which assumes 0,1.
Otherwise call it ```shape_check_01``` and the next time we need one for 2,3 will call it ```shape_check_23```.
Don't.

Reply:
We do it because we must (Mask R-CNN), not because we like it. Done, it's just a few lines of code anyway.